src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -624,13 +624,8 @@ private static (uint userId, uint groupId) GetUserAndGroupIds(ProcessStartInfo s
         private unsafe static (uint? userId, uint? groupId) GetUserAndGroupIds(string userName)
         {
             Interop.Sys.Passwd? passwd;
-#if DEBUG
-            // Use an artificially small buffer in DEBUG to test the fallback path
-            const int BufLen = 2;
-#else
             // First try with a buffer that should suffice for 99% of cases.
             const int BufLen = 1024;
```

Comment:
1K seems like a large stackalloc amount, but I don't have any concrete data to back that up.

Reply:
It is the same we are using in the two places we are calling `getpwuid_r`:

https://github.com/dotnet/corefx/blob/bfac45af5dddcbd70b6935fcf92b4b8755f3cd1c/src/Common/src/System/IO/PersistedFiles.Unix.cs#L100

https://github.com/dotnet/corefx/blob/6f5ce856e9f40a7d247c241a6bb916fc546045ff/src/System.Runtime.Extensions/src/System/Environment.Unix.cs#L371

Reply:
Yes.  In general it seems like a large amount :)

Reply:
Logged https://github.com/dotnet/corefx/issues/26959

Reply:
I put notes to #26959.  We should be able to get the limit from OS - at least on Linux and OSX.